lib/ansible/modules/monitoring/uptimerobot.py

```diff
@@ -27,124 +27,167 @@
+            if state == self.params['status']:
+                self.api_method = method
+
+    def get_monitor_id(self, fname):
+        monitors = self.get_monitors()
+        for m_dict in monitors['monitors']:
+            if m_dict['friendly_name'] == fname:
+                mid = m_dict['id']
+                return mid
+
+    def get_monitors(self):
+        self.body = {}
+        self.body['api_key'] = self.params['api_key']
+        self.body = json.dumps(self.body)
+        self.uri = API_BASE + "getMonitors"
+        state = self.api_call()
+        return state
+
+    def api_call(self):
+        req = Request()
+        state = req.post(url=self.uri, data=self.body, headers=self.headers)
+        json_response = json.loads(to_text(state.read(), errors='surrogate_or_strict'))
+        return json_response
+
+    def api_request(self):
+        self.uri = API_BASE + self.api_method
+        state = self.api_call()
+        return state, self.api_method
+
+    def check_dict(self, dictionary):
+        tmp = {}
+        for k, v in dictionary.items():
+            if v != '':
+                tmp[k] = v
+        if dictionary['status'] != 'created':
+            tmp['id'] = self.get_monitor_id(dictionary['friendly_name'])
+        dictionary.clear()
+        return tmp
 
 
 def main():
+    result = dict(
+        changed=False,
+        original_message='',
+        message=''
+    )
+
+    api_methods = dict(
+        started='editMonitor',
+        present='newMonitor',
+        paused='editMonitor',
+        absent='deleteMonitor',
+        created='newMonitor'
+    )
 
     module = AnsibleModule(
         argument_spec=dict(
-            state=dict(required=True, choices=['started', 'paused']),
+            name=dict(required=True),
+            url=dict(required=True),
+            check_type=dict(required=False, choices=['http', 'dns']),
+            state=dict(required=True, choices=['started', 'paused', 'absent', 'created', 'present']),
             apikey=dict(required=True, no_log=True),
-            monitorid=dict(required=True)
+            monitorid=dict(required=False)
         ),
         supports_check_mode=SUPPORTS_CHECK_MODE
     )
 
     params = dict(
-        apiKey=module.params['apikey'],
-        monitors=module.params['monitorid'],
+        api_key=module.params['apikey'],
         monitorID=module.params['monitorid'],
+        friendly_name=module.params['name'],
+        url=module.params['url'],
+        type=module.params['check_type'],
+        status=module.params['state'],
         format=API_FORMAT,
         noJsonCallback=API_NOJSONCALLBACK
     )
 
-    check_result = checkID(module, params)
+    uprobot = UptimeRobot(params, api_methods)
+    state, api_method = uprobot.api_request()
+    result['message'] = state['stat']
+    result['original_message'] = state
+    if result['message'] != 'ok':
+        module.fail_json(msg='Could not perform action {0}'.format(api_method))
```

Comment:
I am not able to create a new monitor -

```
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "apikey": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "check_type": null,
            "monitorid": null,
            "name": "myMonitor_002",
            "state": "present",
            "url": "http://www.my-domain.com"
        }
    },
    "msg": "Could not perform action newMonitor"
}
``

Reply:
Interesting, I can create and delete.
I use a role with one task like this:
```
---
- name: create Monitor
  custom_uptimerobot:
    state: absent
    url: https://www.google.ch
    name: TEST
    apikey: "MY_KEY"
  delegate_to: localhost
...
```
And a Playbook like this:
```
---
- hosts:
  - somehost
  roles:
    - uptimerobot
  gather_facts: no
...
```

Reply:
@Akasurde Try if it works if you remove the null values. The `check_type` and `monitorid` keys can be ignored because they are optional.